components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -216,20 +216,41 @@ open class GleanInternalAPI internal constructor () {
         val newClientId = uuidClientInfoSnapshot?.get("client_id")
         if (newClientId == null) {
             val uuidPingInfoSnapshot = UuidsStorageEngine.getSnapshot("glean_ping_info", false)
-            uuidPingInfoSnapshot?.get("client_id")?.let {
-                UuidsStorageEngine.record(GleanInternalMetrics.clientId, it)
+            val legacyClientId = uuidPingInfoSnapshot?.get("client_id")
+            if (legacyClientId != null) {
+                UuidsStorageEngine.record(GleanInternalMetrics.clientId, legacyClientId)
+            } else {
+                // Apparently, this is a very old build that was already run
+                // at least once and has a different name for the underlying
+                // UUID storage engine persistence (i.e. UuidStorageEngine.xml
+                // vs using the full class name with the namespace).
+                // This should be mostly devs and super early adopters, so just regenerate
+                // the data.
+                UuidsStorageEngine.record(GleanInternalMetrics.clientId, UUID.randomUUID())
```

Comment:
Random thought -- we could record an error on the metric here so we could see how often this happens.  If it shows up again in the future, it might help us detect mistakes...

Reply:
Ooh, I like this idea!

Reply:
Yeah, that might be a good idea. The thing is that, in contrast to desktop, we have no alerting on this _for now_. It it would be up to humans (that's us!) to manually check in that for now. I'd rather have a monitor or alerting on schema validation errors (that caught this!!!!), which is a lower hanging fruit with the tooling that's available right now. Additionally, I'm planning on prioritizing the integration test.